Move theming support from Beta to Main

jordanjones243 opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #223, #234

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

<!-- Generated by sourcery-ai[bot]: start summary -->

## Summary by Sourcery

Move theming support from Beta to Main by introducing theme support for the auro-dropdown component, refactoring color token structures, and updating to use the beta version of auro-icon with theme support. Remove deprecated OnDark API references and update documentation to reflect these changes.

New Features:
- Introduce theme support for the auro-dropdown component, allowing customization through design tokens.

Enhancements:
- Refactor the auro-dropdown component to use a new color token structure with tier 3 tokens.
- Update the auro-dropdown component to use the beta version of auro-icon with theme support.
- Replace hard-coded size values with design tokens for better maintainability and consistency.

Documentation:
- Add documentation for theme support in the auro-dropdown component, including examples of how to restyle the component using design tokens.

Chores:
- Remove all references to the OnDark API, simplifying the codebase and removing deprecated styles.

<!-- Generated by sourcery-ai[bot]: end summary -->

jordanjones243 committed:
build: update design tokens and all other dependencies

jordanjones243 committed:
feat: refactor color token structure with tier 3 tokens #223

jordanjones243 committed:
perf: replace hard coded size values with tokens

jordanjones243 committed:
refactor: remove OnDark style support

BREAKING CHANGE: remove all references to the OnDark API

jordanjones243 committed:
perf(icon): update to use beta of auro-icon with theme support

semantic-release-bot committed:
chore(release): 3.0.0-beta.1 [skip ci]

# [3.0.0-beta.1](https://github.com/AlaskaAirlines/auro-dropdown/compare/v2.10.6...v3.0.0-beta.1) (2024-06-24)

### Code Refactoring

* remove OnDark style support ([1557e77](https://github.com/AlaskaAirlines/auro-dropdown/commit/1557e77d4aa819b8d276d38548865a9553959604))

### Features

* refactor color token structure with tier 3 tokens [#223](https://github.com/AlaskaAirlines/auro-dropdown/issues/223) ([14168c1](https://github.com/AlaskaAirlines/auro-dropdown/commit/14168c184ce7c75341378611930b758d15c22fb5))

### Performance Improvements

* **icon:** update to use beta of auro-icon with theme support ([b16fe03](https://github.com/AlaskaAirlines/auro-dropdown/commit/b16fe03630488922c708f69bf69e39e918fc5473))
* replace hard coded size values with tokens ([d35b4be](https://github.com/AlaskaAirlines/auro-dropdown/commit/d35b4be4872d6f0e88188c435713f118338f10e3))

### BREAKING CHANGES

* remove all references to the OnDark API

jordanjones243 committed:
perf: update to use library scripts and utilities

semantic-release-bot committed:
chore(release): 3.0.0-beta.2 [skip ci]

# [3.0.0-beta.2](https://github.com/AlaskaAirlines/auro-dropdown/compare/v3.0.0-beta.1...v3.0.0-beta.2) (2024-08-13)

### Performance Improvements

* update to use library scripts and utilities ([3ae5a3f](https://github.com/AlaskaAirlines/auro-dropdown/commit/3ae5a3f6dc96a815290b1f6e1ff8c45214c8ca1a))

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AlaskaAirlines/auro-dropdown?pullRequest=236) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/AlaskaAirlines/auro-dropdown?pullRequest=236) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: jordanjones243<br/>:x: semantic-release-bot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AlaskaAirlines/auro-dropdown?pullRequest=236) it.</sub>

jordanjones243 committed:
perf: update icon version to new beta release

jordanjones243 committed:
fix: use auro-icon attribute selector

semantic-release-bot committed:
chore(release): 3.0.0-beta.3 [skip ci]

# [3.0.0-beta.3](https://github.com/AlaskaAirlines/auro-dropdown/compare/v3.0.0-beta.2...v3.0.0-beta.3) (2024-08-21)

### Bug Fixes

* use auro-icon attribute selector ([d9e3eec](https://github.com/AlaskaAirlines/auro-dropdown/commit/d9e3eec868e3617312125d0853ab6f1a003fa0a5))

### Performance Improvements

* update icon version to new beta release ([9d68d36](https://github.com/AlaskaAirlines/auro-dropdown/commit/9d68d3604c7936570d0da0c96cbb744e5b4cbb42))

jordanjones243 committed:
perf: update icon and all other dependencies

semantic-release-bot committed:
chore(release): 3.0.0-beta.4 [skip ci]

# [3.0.0-beta.4](https://github.com/AlaskaAirlines/auro-dropdown/compare/v3.0.0-beta.3...v3.0.0-beta.4) (2024-08-26)

### Performance Improvements

* update icon and all other dependencies ([4cef867](https://github.com/AlaskaAirlines/auro-dropdown/commit/4cef867c48cb27631e3651166ebd0fea5598de38))

jordanjones243 committed:
build: update dependencies

jordanjones243 committed:
fix(theming): resolve minor color theming issues #223

semantic-release-bot committed:
chore(release): 3.0.0-beta.5 [skip ci]

# [3.0.0-beta.5](https://github.com/AlaskaAirlines/auro-dropdown/compare/v3.0.0-beta.4...v3.0.0-beta.5) (2024-09-17)

### Bug Fixes

* **theming:** resolve minor color theming issues [#223](https://github.com/AlaskaAirlines/auro-dropdown/issues/223) ([bb61dcb](https://github.com/AlaskaAirlines/auro-dropdown/commit/bb61dcb54f9f60a319dd30dabde09153b4a9607f))

jordanjones243 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,35 @@
+
+### BREAKING CHANGES
+
+* remove all references to the OnDark API
```
The OnDark API was never advertised to users.

jordanjones243 committed:
perf: refactor custom component registration config

jordanjones243 committed:
docs: remove beta releases from CHANGELOG

semantic-release-bot committed:
chore(release): 3.0.0-beta.6 [skip ci]

# [3.0.0-beta.6](https://github.com/AlaskaAirlines/auro-dropdown/compare/v3.0.0-beta.5...v3.0.0-beta.6) (2024-09-18)

### Performance Improvements

* refactor custom component registration config ([1ec40bc](https://github.com/AlaskaAirlines/auro-dropdown/commit/1ec40bc2a7b75739eb2626b571ee34932f1d2f31))

jordanjones243 committed:
docs: remove beta from changelog

jordanjones243 committed:
perf: update dependencies

BREAKING CHANGE: trigger major release for theming support #223

semantic-release-bot committed:
chore(release): 3.0.0-beta.7 [skip ci]

# [3.0.0-beta.7](https://github.com/AlaskaAirlines/auro-dropdown/compare/v3.0.0-beta.6...v3.0.0-beta.7) (2024-09-24)

### Performance Improvements

* update dependencies ([3098c34](https://github.com/AlaskaAirlines/auro-dropdown/commit/3098c34f7ae2db13928e27505eb3e0b86abc2a60))

### BREAKING CHANGES

* trigger major release for theming support #223

jordanjones243 committed:
docs: remove beta release from changelog

blackfalcon commented:
:tada: This PR is included in version 3.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@aurodesignsystem/auro-dropdown/v/3.0.0)
- [GitHub release](https://github.com/AlaskaAirlines/auro-dropdown/releases/tag/v3.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
